cmd/component/model.go

```diff
@@ -0,0 +1,228 @@
+// and its affiliates.
+// The copyright notice above does not evidence any actual or intended
+// publication of such source code.
+
+package component
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss/table"
+	"github.com/datarobot/cli/internal/copier"
+	"github.com/datarobot/cli/tui"
+)
+
+type (
+	errMsg              struct{ err error }
+	screens             int
+	componentsLoadedMsg struct {
+		components []copier.Component
+	}
+)
+
+const (
+	// TODO: Add more keys as we start to use them in upcoming work
+	keyExit  = "esc"
+	keyEnter = "enter"
+)
+
+const (
+	listScreen = screens(iota)
+	updateScreen
+)
+
+type Model struct {
+	err            error
+	infoMessage    string
+	screen         screens
+	initialScreen  screens
+	components     []copier.Component
+	updateFileName string
+}
+
+func NewComponentModel(initialScreen screens) Model {
+	return Model{
+		screen:        initialScreen,
+		initialScreen: initialScreen,
+	}
+}
+
+func NewUpdateComponentModel(currentScreen screens, updateFileName string) Model {
+	return Model{
+		screen:         currentScreen,
+		initialScreen:  updateScreen, // Always set initial screen to update
+		updateFileName: updateFileName,
+	}
+}
+
+func (m Model) Init() tea.Cmd {
+	switch m.initialScreen {
+	case listScreen:
+		return tea.Batch(m.loadComponents(), tea.WindowSize())
+	case updateScreen:
+		if m.updateFileName == "" {
+			return tea.Batch(m.loadComponents(), tea.WindowSize())
+		}
+
+		return tea.WindowSize()
+	}
+
+	return tea.WindowSize()
+}
+
+func (m Model) loadComponents() tea.Cmd {
+	return func() tea.Msg {
+		answers, err := copier.AnswersFromPath(".")
+		if err != nil {
+			return errMsg{err}
+		}
+
+		components, err := copier.ComponentsFromAnswers(answers)
+		if err != nil {
+			return errMsg{err}
+		}
+
+		// If we've found zero components return error message that is handled by UI
+		if len(components) == 0 {
+			return errMsg{errors.New("No components were found.")} //nolint:revive,staticcheck
+		}
+
+		return componentsLoadedMsg{components}
+	}
+}
+
+func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) { //nolint:cyclop
+	switch msg := msg.(type) {
+	case componentsLoadedMsg:
+		m.components = msg.components
+		if m.initialScreen == updateScreen && m.updateFileName == "" {
+			m.infoMessage = "Please use an 'answer file' value with 'dr update <answer_file>'."
+		}
+	}
+
+	switch m.screen {
+	case listScreen:
+		switch msg := msg.(type) {
+		case tea.KeyMsg:
+			switch keypress := msg.String(); keypress {
+			case keyExit:
+				return m, tea.Quit
+			}
+		case errMsg:
+			m.err = msg.err
+			return m, nil
+		}
+	case updateScreen:
+		switch msg := msg.(type) {
+		case tea.KeyMsg:
+			switch keypress := msg.String(); keypress {
+			case keyExit:
+				return m, tea.Quit
+			case keyEnter:
+				if m.updateFileName != "" {
+					err := runUpdate(m.updateFileName)
+					if err != nil {
+						m.err = err
+					}
```

Comment:
If it's out of scope of this PR - comment it out for now:

```suggestion
					// TODO: run copier.ExecUpdate using tea.ExecProcess here

```

Reply:
I might do that - thanks.